src/generators.py

```diff
@@ -0,0 +1,89 @@
+from typing import List, Dict, Iterator, Any
+
+
+def filter_by_currency(transactions: List[Dict[str, Any]], currency: str) -> Iterator[Dict[str, Any]]:
+    # Генератор фильтрует транзакции по валюте
+    return (transaction for transaction in transactions
```

Comment:
Если ключа нет - значит это не транзация и лучше упасть с ошибкой
```suggestion
    return (t for t in transactions if t['operationAmoun']['currency']['code'] == currency)
```